eyelid-match-ops/src/primitives/yashe.rs

```diff
@@ -261,53 +249,69 @@ where
         R: SampleRange<T> + Clone,
         C::Coeff: From<T>,
     {
-        // TODO: use Poly::coeffs_modify_include_zero() here and benchmark
         let mut res = Poly::non_canonical_zeroes(C::MAX_POLY_DEGREE);
-        for i in 0..C::MAX_POLY_DEGREE {
+        Poly::coeffs_modify_include_zero(&mut res, |coeff: &mut <C as PolyConf>::Coeff| {
             let coeff_rand = rng.gen_range(range.clone());
-            res[i] = coeff_rand.into();
-        }
+            *coeff = coeff_rand.into();
+        });
 
-        // Raw coefficient access must be followed by a truncation check.
-        res.truncate_to_canonical_form();
         res
     }
 
-    /// Plaintext addition is trivial
+    /// Sample a polynomial with random binnary coefficients, i.e. 0, 1
+    pub fn sample_binary_message(&self, rng: &mut ThreadRng) -> Message<C> {
```

Comment:
This method already exists here:
https://github.com/Inversed-Tech/eyelid/blob/ae61658ad1e5372e003cdf545f9cb2b181f4dff1/eyelid-match-ops/src/primitives/yashe/test.rs#L52

What is stopping you from calling it?
It should be available to all tests and benchmarks.

Reply:
I removed the testing method.